fixture-monkey-api/src/main/java/com/navercorp/fixturemonkey/api/type/Types.java

```diff
@@ -153,31 +155,39 @@ public static List<AnnotatedType> getGenericsTypes(AnnotatedType annotatedType)
 	public static AnnotatedType resolveWithTypeReferenceGenerics(AnnotatedType ownerType, Field field) {
 		if (!(ownerType instanceof AnnotatedParameterizedType)) {
 			AnnotatedType fieldAnnotatedType = field.getAnnotatedType();
-			if (TypeVariable.class.isAssignableFrom(fieldAnnotatedType.getType().getClass())) {
-				return new AnnotatedType() {
-					@Override
-					public Type getType() {
-						return Object.class;
-					}
-
-					@Override
-					public <T extends Annotation> T getAnnotation(Class<T> annotationClass) {
-						return fieldAnnotatedType.getAnnotation(annotationClass);
-					}
-
-					@Override
-					public Annotation[] getAnnotations() {
-						return fieldAnnotatedType.getAnnotations();
-					}
 
-					@Override
-					public Annotation[] getDeclaredAnnotations() {
-						return fieldAnnotatedType.getDeclaredAnnotations();
-					}
-				};
-			} else {
-				return fieldAnnotatedType;
+			AnnotatedType annotatedSuperClassType = ((Class<?>)ownerType.getType()).getAnnotatedSuperclass();
+			if (annotatedSuperClassType != null) {
+				if (ParameterizedType.class.isAssignableFrom(annotatedSuperClassType.getType().getClass())) {
+					ParameterizedType parameterizedType = (ParameterizedType)annotatedSuperClassType.getType();
+					Type[] actualTypeArguments = parameterizedType.getActualTypeArguments();
+
+					int index = getTypeVariableIndex((TypeVariable<?>)field.getGenericType());
+					Type actualType = actualTypeArguments[index];
```

Comment:
index 가 없는경우 에러가 날거 같습니다 처리를 해주면 좋을거 같습니다.

Reply:
@flex-myeonghyeon 
https://docs.oracle.com/javase/7/docs/api/java/lang/reflect/ParameterizedType.html
```
Note that in some cases, the returned array be empty. This can occur if this type represents a non-parameterized type nested within a parameterized type.
```
문서를 살펴보면 index가 없는 경우는 nested 타입인 경우라고 합니다.
테스트해보았을 때 아직까지 Fixture Monkey에서는 nested 클래스인 경우 상위 클래스가 파라미터로 들어오는 문제가 있어서 정상적으로 생성할 수 없습니다.

말씀하신 처리는 다음 이슈를 처리할 때 추가하면 어떨까요?
https://github.com/naver/fixture-monkey/issues/548

Reply:
@seongahjo 제 의미는 `아래 메소드에서 index 가 -1 을 반환할 경우 배열에서 `actualTypeArguments[-1]` 에서 에러가 날거 같다` 였습니다.

Reply:
@mhyeon-lee 
actualTypeArguments이 비어있지 않으면 `-1`이 반환하는 경우가 없을 것 같다고 생각하긴 했습니다.
-1보다는 말씀하신 것처럼 Optional로 표현하는 게 더 의미가 잘 나타나겠네요. 반영하겠습니당.